src/main/java/seedu/address/logic/commands/edit/EditCommand.java

```diff
@@ -90,8 +91,15 @@ public CommandResult execute(Model model) throws CommandException {
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+        boolean phoneExists = model.hasPhone(editedPerson);
+        boolean emailExists = model.hasEmail(editedPerson);
```

Comment:
might want to rename these variables, `phoneExists` can mean inside the `Person` or inside the `Model`

so maybe something like `phoneIsFoundInModel` or list or something along those lines, and same for email